File: src/main/java/no/nav/pensjon/regler/domain/beregning2011/Garantipensjon.java
package no.nav.pensjon.regler.domain.beregning2011;

import no.nav.pensjon.regler.domain.beregning.Ytelseskomponent;
import no.nav.pensjon.regler.domain.kode.YtelsekomponentTypeCti;

public class Garantipensjon extends Ytelseskomponent {
    private static final long serialVersionUID = 1495797783492118283L;

    public Garantipensjon() {
        super();
        ytelsekomponentType = new YtelsekomponentTypeCti("GAP");
    }

    public Garantipensjon(Garantipensjon garp) {
        super(garp);
        ytelsekomponentType = new YtelsekomponentTypeCti("GAP");
    }
}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning2011/Garantipensjon.kt
package no.nav.pensjon.regler.domain.beregning2011

import no.nav.pensjon.regler.domain.beregning.Ytelseskomponent
import no.nav.pensjon.regler.domain.enum.YtelseskomponentTypeEnum

class Garantipensjon : Ytelseskomponent() {
    override val ytelsekomponentTypeEnum: YtelseskomponentTypeEnum = YtelseskomponentTypeEnum.GAP
}